pyirf/interpolation.py

```diff
@@ -0,0 +1,99 @@
+"""Functions for performing interpolation of IRF to the values read from the data."""
+
+import numpy as np
+import astropy.units as u
+from scipy.interpolate import griddata
+
+
+def compare_irf_cuts(cuts):
+    """
+    checks if the same cuts have been applied in all of IRFs
+
+    Parameters
+    ----------
+    cuts: list of QTables
+        list of cuts each entry in the list correspond to one set of IRFs
+    Returns
+    -------
+    match: Boolean
+        if the cuts are the same in all the files
+    """
+
+    for i in range(len(cuts)-1):
+        if (cuts[i] != cuts[i+1]).any():
+            raise ValueError("difference in cuts")
+    return True
+
+
+def interpolate_effective_area_per_energy_and_fov(effective_area, grid_points, target_point, min_effective_area=1. * u.Unit('m2'), method='linear'):
+    """
+    Takes a grid of effective areas for a bunch of different parameters
+    and interpolates (log) effective areas to given value of those parameters
+
+    Parameters
+    ----------
+    effective_area: np.array of astropy.units.Quantity[area]
+        grid of effective area, of shape (n_grid_points, n_fov_offset_bins, n_energy_bins)
+    grid_points: np.array
+        list of parameters corresponding to effective_area, of shape (n_grid_points, n_interp_dim)
+    target_point: np.array
+        values of parameters for which the interpolation is performed, of shape (n_interp_dim)
+    min_effective_area: astropy.units.Quantity[area]
+        Minimum value of effective area to be considered for interpolation
+    method: 'linear’, ‘nearest’, ‘cubic’
+        Interpolation method
+
+    Returns
+    -------
+    aeff_interp: astropy.units.Quantity[area]
+        Interpolated Effective area array with shape (n_energy_bins, n_fov_offset_bins)
+    """
+
+    # get rid of units
+    effective_area = effective_area.to_value(u.m**2)
+    min_effective_area = min_effective_area.to_value(u.m**2)
+
+    # remove zeros and log it
+    effective_area[effective_area < min_effective_area] = min_effective_area
+    effective_area = np.log(effective_area)
```

Comment:
Why do we interpolate effective area in logspace? The only axis over which effective area is expected to vary over multiple orders of magnitude is energy, but thats not the interpolation axis. 

Wouldn't normal linear interpolation make more sense for different values of e.g. azimuth / zenith?

Reply:
the reason is mostly the threshold effects - when you interpolate the energy bins close to the threshold, the zenith angle shifts the whole curve in energy scale and this means that the difference in zenith can produce a large difference in collection area and thus it is safer to interpolate such points in log space.

Reply:
Ok, let's merge it like this and maybe we need to make this optional and check on simulations which implementation scheme performs better.

Reply:
I think interpolation in the logA logE space should work well at all energies, not just at threshold, so for simplicity I would keep (at least as default) the log-log approach everywhere.

Reply:
But energy is not used in the interpolation at all. The interpolation happens for all energy bins independently.

And then it boils down, what kind of behaviour you expect of effective area vs. the changing (interpolation) variable. 
That would just needed to be checked for the specific interpolation one is doing.

Reply:
Ah, right, the interpolation in energy is done by gammapy.
I think Julian's point that logA is also better for interpolation in zenith is probably true, but there is not other way of knowing that testing it.